app/scripts/directives/multiSelectMenu.js

```diff
@@ -58,17 +58,17 @@ angular.module('gsUiInfraApp')
                 }
 
                 /**
-                 * Bind data to ng-model
+                 * call onchange function when model change
                  */
-                $scope.$watch('selected', function (newValue) {
-                    if (newValue) {
-
-                        setValue(newValue);
-                        if(angular.isFunction($scope.onchange)) {
-                            $scope.onchange({filter: newValue});
+                if(angular.isFunction($scope.onchange)) {
```

Comment:
onchange is a function injected to the directive isolated scope. I want to make sure, the user do wish to subscribe to this event ( by injecting a function) , before I'm watching the model for nothing.

So I am checking If I am given a function.
Why wouldn't it work?